derive-macros/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+use proc_macro2::{Ident, TokenStream};
+use quote::{quote, quote_spanned};
+use syn::spanned::Spanned;
+use syn::{parse_macro_input, Data, DataStruct, DeriveInput, Fields, PathArguments, Type};
+
+#[proc_macro_derive(Schema, attributes(schema))]
+pub fn derive_schema(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+    let struct_ident = input.ident;
+
+    let schema_fields = gen_schema_fields(&input.data);
+    let output = quote! {
+        impl crate::actions::schemas::GetField for #struct_ident {
+            fn get_field(name: impl Into<String>) -> crate::schema::StructField {
+                use crate::actions::schemas::GetField;
+                crate::schema::StructField::new(
+                    name,
+                    crate::schema::StructType::new(vec![
+                        #schema_fields
+                    ]),
+                    // By default not nullable. To make something nullable wrap it in an Option
+                    false,
+                )
+            }
+        }
+    };
+    proc_macro::TokenStream::from(output)
+}
+
+// turn our struct name into the schema name, goes from snake_case to camelCase
```

Comment:
I don't know where to put the doc comment, but somewhere we should be careful to explain that the actual field names are all mandated by Delta spec, and so the user of this macro is responsible to ensure that e.g. `Metadata::schema_string` is the snake-case-ified version of `schemaString` from Delta's [Change Metadata](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#change-metadata) action, in order to keep rust happy. This macro is written with the assumption that it merely undoes that (previously correctly performed) transformation.

The same explains why it's ok to use `to_ascii_uppercase` below -- all Delta field names are plain ASCII.